python-checks/src/main/java/org/sonar/python/checks/UnusedLocalVariableCheck.java

```diff
@@ -120,17 +123,36 @@ private static boolean isVariableAccessedInStringTemplate(Symbol symbol, StringL
 
   public PreciseIssue createIssue(SubscriptionContext ctx, Symbol symbol, Usage usage) {
     if (isSequenceUnpacking(usage)) {
-      var quickFix = PythonQuickFix.newQuickFix(SEQUENCE_UNPACKING_QUICK_FIX_MESSAGE, TextEditUtils.replace(usage.tree(), "_"));
+      var quickFix = PythonQuickFix.newQuickFix(RENAME_QUICK_FIX_MESSAGE, TextEditUtils.replace(usage.tree(), "_"));
       var issue = ctx.addIssue(usage.tree(), String.format(SEQUENCE_UNPACKING_MESSAGE, symbol.name()));
       issue.addQuickFix(quickFix);
       return issue;
+    } else if (isLoopIndex(usage, symbol)) {
+      PreciseIssue issue = ctx.addIssue(usage.tree(), String.format(LOOP_INDEX_MESSAGE, symbol.name()));
+      PythonQuickFix quickFix = PythonQuickFix.newQuickFix(RENAME_QUICK_FIX_MESSAGE, TextEditUtils.replace(usage.tree(), "_"));
```

Comment:
It probably makes more sense to instantiate the quick fix only if we're going to apply it (i.e in the `foundUnderscoreSymbol == null` branch).